### PR TITLE
Run automated tests when lua files change

### PR DESCRIPTION
**Pull request by x2048**
Expands Lua workflow to include:
* automated testing of devtest game
* linting of devtest game

Based on conversation in [IRC](https://irc.minetest.net/minetest-dev/2022-04-07#i_5958137)

## To do

This PR is Ready for Review.

## How to test

* Github workflows must pass

### CODE REVIEW

**Comment by SmallJoker** [comment]
`.luacheckrc` for `devtest` is needed too. Here are a few templates for the API names:

 * https://github.com/minetest/minetest/blob/master/.luacheckrc
 * https://github.com/minetest-mods/technic/blob/master/.luacheckrc

**Comment by x2048** [comment]
@sfan5 @SmallJoker I've added .luacheckrc for devtest, so that the job passes now. I guess, fixing the warnings (and removing suppressions from the config file) is a different PR?

File: games/devtest/.luacheckrc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ignore = {
+	"131", -- Unused global variable
+	"211", -- Unused local variable
+	"231", -- Local variable never accessed
+	"311", -- Value assigned to a local variable is unused
+	"412", -- Redefining an argument
+	"421", -- Shadowing a local variable
+	"431", -- Shadowing an upvalue
+	"432", -- Shadowing an upvalue argument
+	"611", -- Line contains only whitespace
```
Comment (sfan5) [inline games/devtest/.luacheckrc:15]: only whitespace? that sounds worth fixing

**Comment by sfan5** [comment]
Fine by me to do it in a different PR.

**Comment by x2048** [comment]
I can merge the PR as soon as it gets approved

**Review by sfan5 — APPROVED** [review]

**Pull request merged**